src/server/router/electricity.router.ts

```diff
@@ -1,19 +1,42 @@
 import { createRouter } from './context';
-import { unregisteredElectricityRequestSchema } from '../../schema/electricity.schema';
+import {
+  electricityRequestSchema,
+  ElectricityResponse,
+} from '../../schema/electricity.schema';
 import { carbonEstimateRequest } from '../../utils/apiHelpers';
 
 export const electricityRouter = createRouter().mutation(
-  'unregistered-request-electricity',
+  'calculation-request',
   {
-    input: unregisteredElectricityRequestSchema,
-    async resolve({ input }) {
+    input: electricityRequestSchema,
+
+    async resolve({ input, ctx }) {
+      const user = ctx.session?.user;
       const response = await fetch(
         carbonEstimateRequest(JSON.stringify({ type: 'electricity', ...input }))
       );
 
-      return response.json().then((data) => {
-        return data.data.attributes;
-      });
+      let carbonResponse: ElectricityResponse = await response
+        .json()
+        .then((data) => {
+          return data.data.attributes;
+        });
+
+      if (user) {
+        const updateUser = await ctx.prisma.electricityUse
+          .create({
+            data: {
+              userId: user.id,
+              electricity_value: carbonResponse.electricity_value,
+              electricity_unit: carbonResponse.electricity_unit,
+              estimated_at: carbonResponse.estimated_at,
+              carbon_g: carbonResponse.carbon_g,
+            },
+          })
+          .then((response) => response.userId);
+        updateUser;
```

Comment:
Does this line do anything?

Reply:
I think this was a remnant when I was testing to see if it worked, it was essentially a `console.log`, I removed it!